domain/model/service/modelservice.go

```diff
@@ -145,15 +161,23 @@ func (s *ModelService) GetModelMetrics(ctx context.Context) (coremodel.ModelMetr
 func (s *ModelService) CreateModel(
 	ctx context.Context,
 	controllerUUID uuid.UUID,
+	agentVersion version.Number,
```

Comment:
As per above I think if we have code in this func that specifically deals with the fact version can be zero we should just remove this as an argument. Model migration import will need to supply this information but we should introduce a new function signature just for that operation.

Otherwise it just seems like we are bleeding out one requirement to all other callers.